### ISSUE TITLE
Quantize doesn't reduce the GPU memory usage 

### ISSUE THREAD

**Issue by Judy0406-CJ** [huggingface/text-generation-inference#1300]
### System Info

Latest docker

### Information

- [X] Docker
- [ ] The CLI directly

### Tasks

- [ ] An officially supported command
- [ ] My own modifications

### Reproduction

It used 60G GPU memory on my A100 server to run a model without `quantize`.  Adding `quantize` doesn't help reduce the memory.

### Expected behavior

Memory can go down.

**Comment by mapa17** [comment]
Are you loading a quantized version of the same model?

**Comment by Judy0406-CJ** [comment]
@mapa17 I don't think it is a quantized version. But [text-generation-webui](https://github.com/oobabooga/text-generation-webui) can dramatically reduce the VRAM for same model when I enable the quantization.

**Comment by hdnh2006** [comment]
> It's because we allocate the rest of the GPU VRAM for the KV cache. TGI will always use all available memory as this dramatically improves throughput. If you want to limit the amount of memory taken by TGI, you can use `--cuda-memory-fraction 0.5` for example (50%).


Thanks for the answer, can you please explain a bit more your comment? For example, what does KV cache means?

Thank you

**Comment by suparious** [comment]
> what does KV cache means

This refers to the amount memory that the inference uses for your history and context. By default, it prefers to use the maximum supported by the model or that is available to the system.

So even with a Quant model, the inference engine will use the extra memory to optimize for better context handling.
